### PR TITLE
Support microArch

### PR DESCRIPTION
**Pull request by renfeiw**
- update EnvDetector and MachineInfo to support microArch
- add detection for skylake
- add parsing logic in TKG to check microArch
- update corresponding make files

related: #336

Signed-off-by: renfeiw <renfeiw@ca.ibm.com>

### CODE REVIEW

**Comment by renfeiw** [comment]
modified openj9 testExample to test tkg on skylake: Grinder/26678, not on skylake: Grinder/26679

File: src/org/testKitGen/Arguments.java [hunk 1]
```diff
@@ -68,7 +69,9 @@ enum Mode {
 			+ "    --testTime=<number>       Specify expected length of test running time (minutes) on each machines for mode parallelList, this option will be suppressed if numOfMachines is given\n"
 			+ "                              If testTime and numOfMachines are not provided, default numOfMachines will be used\n"
 			+ "    --TRSSURL=<serverURL>     Specify the TRSS server URL for mode parallelList\n"
-			+ "                              Defaults to " + Constants.TRSS_URL + "\n";
+			+ "                              Defaults to " + Constants.TRSS_URL + "\n"
+			+ "    --microArch=<microArch>   Specify micro-architeture\n"
```
Comment (smlambert) [inline src/org/testKitGen/Arguments.java:73]:
```suggestion
			+ "    --microArch=<microArch>   Specify micro-architecture\n"
```

**Review by smlambert — APPROVED** [review]
LGTM thanks!  (with a minor typo in usage comment as a suggested change)

File: src/org/testKitGen/Arguments.java (hunk 1 above)
Comment (renfeiw) [reply src/org/testKitGen/Arguments.java:73]: Thank you! Updated.

**Comment by llxia** [comment]
Please add the readme with examples and how to add additional microArch support. It can be in another PR.

**Comment by renfeiw** [comment]
> Please add the readme with examples and how to add additional microArch support. It can be in another PR.

Sure. I will do it in another PR in openj9 repo.

**Review by llxia — APPROVED** [review]
LGTM

**Pull request merged**